#VRML_SIM V6.0 utf8
WorldInfo {
  info [
    "Description"
    "Author: first name last name <e-mail>"
    "Date: DD MMM YYYY"
  ]
  runRealTime TRUE
}
Viewpoint {
  orientation 0.375609 -0.84674 -0.376763 4.68361
  position 1.16305 1.41799 0.400941
}
Background {
  skyColor [
    0.4 0.7 1
  ]
}
PointLight {
  ambientIntensity 0.6
  intensity 0.6
  location 0.75 0.5 0.5
  radius 10
}
PointLight {
  ambientIntensity 0.6
  intensity 0.6
  location -0.5 0.5 0.35
  radius 10
}
PointLight {
  ambientIntensity 0.6
  intensity 0.6
  location 0.45 0.5 -0.5
  radius 10
}
PointLight {
  ambientIntensity 0.6
  intensity 0.6
  location -0.5 0.5 -0.35
  radius 10
}
Solid {
  translation -0.5 0 -0.5
  children [
    Shape {
      geometry ElevationGrid {
        color Color {
          color [
            0 0 0
            1 1 1
          ]
        }
        colorPerVertex FALSE
        xDimension 11
        xSpacing 0.1
        zDimension 11
        zSpacing 0.1
      }
    }
  ]
  boundingObject Shape {
    geometry IndexedFaceSet {
      coord Coordinate {
        point [
          0 0 0
          0 0 1
          1 0 1
          1 0 0
        ]
      }
      coordIndex [
        0, 1, 2, 3, -1
      ]
    }
  }
}
DEF ck DifferentialWheels {
  translation 0.35 0 0.35
  rotation 0 1 0 0.8
  children [
    DistanceSensor {
      translation -0.02 0.063 -0.042
      rotation 0 1 0 2.07
      children [
        DEF infrared Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 1 0 0.501961
                  specularColor 1 0 0.501961
                }
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ir0"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      aperture 1
    }
    DistanceSensor {
      translation 0.02 0.063 -0.042
      rotation 0 1 0 1.07
      children [
        USE infrared
      ]
      name "ir1"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      aperture 1
    }
    DistanceSensor {
      translation -0.045 0.063 0
      rotation 0 1 0 3.07
      children [
        USE infrared
      ]
      name "ir2"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      aperture 1
    }
    DistanceSensor {
      translation 0.045 0.063 0
      rotation 1 0 0 0
      children [
        USE infrared
      ]
      name "ir3"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      aperture 1
    }
    Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 1 0.501961 0.501961
            }
          }
          geometry DEF body Cylinder {
            height 0.08
            radius 0.05
            subdivision 100
          }
        }
      ]
    }
    Solid {
      translation -0.045 0.025 0
      rotation 1 0 0 0
      children [
        DEF wheel Transform {
          rotation 0 0 1 1.57
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 0 0 0
                }
              }
              geometry Cylinder {
                height 0.01
                radius 0.025
              }
            }
          ]
        }
      ]
      name "left_wheel"
    }
    Solid {
      translation 0.045 0.025 0
      rotation 1 0 0 0
      children [
        USE wheel
      ]
      name "right_wheel"
    }
    Receiver {
    }
    GPS {
    }
  ]
  name "ck"
  boundingObject Transform {
    translation 0 0.0415 0
    children [
      USE body
    ]
  }
  controller "lab"
  axleLength 0.09
  wheelRadius 0.025
  speedUnit 0.1
}
DEF wall Solid {
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.501961 0 1
        }
      }
      geometry Extrusion {
        convex FALSE
        crossSection [
          0.5 0.5
          0.5 -0.5
          -0.5 -0.5
          -0.5 0.5
          -0.49 0.5
          -0.49 -0.49
          0.49 -0.49
          0.49 0.49
          -0.4899 0.49
          -0.4899 0.5
          0.5 0.5
        ]
        spine [
          0 0 0
          0 0.1 0
        ]
      }
    }
  ]
  name "wall"
  boundingObject Group {
    children [
      Transform {
        translation 0.495 0.05 0
        children [
          DEF wall_shape Shape {
            appearance Appearance {
              material Material {
                diffuseColor 1 1 1
                specularColor 1 1 1
              }
            }
            geometry Box {
              size 0.01 0.1 1
            }
          }
        ]
      }
      Transform {
        translation -0.495 0.05 0
        children [
          USE wall_shape
        ]
      }
      Transform {
        translation 0 0.05 0.495
        rotation 0 1 0 1.57
        children [
          USE wall_shape
        ]
      }
      Transform {
        translation 0 0.05 -0.495
        rotation 0 1 0 1.57
        children [
          USE wall_shape
        ]
      }
    ]
  }
  locked TRUE
}
Supervisor {
  controller "randoms"
}
LankMark1 {
  translation 0 0.05 -0.45
}
LankMark1 {
  translation 0 0.05 0.45
}
LankMark2 {
  translation 0.45 0.05 0
}
LankMark2 {
  translation -0.45 0.05 0
}
DEF WALL_01 LankMark {
  translation -0.15 0.05 -0.1
  rotation 0 1 0 1.57
}
DEF WALL_02 LankMark {
  translation 0.05 0.05 -0.1
  rotation 0 1 0 1.57
}
